documentation/developer/text/20_env_variables.md

# Environment Variables

There are two files containing environment variables: `.env.dev` for development and `.env.prod` for production. In a production environment, it's recommended to run additional Docker containers, such as one for a database (Postgres in this case). The database container for production has its own set of environment variables, which are included in the `.env.prod.db` file.


## .env.dev

```
DEBUG=1                                            # Show debug information
SECRET_KEY=foo                                     # Dummy value for secret key, used for encrypting passwords
DJANGO_ALLOWED_HOSTS=localhost 127.0.0.1 [::1]     # Allowed hosts set to local hosts
DJANGO_SUPERUSER_PASSWORD=Welcome123               # Set a superuser password to automatically create a superuser (see `tweeti/entrypoint.sh`)
```

__Optional: Set production environment variables for testing (e.g. postgres database)__

```
SQL_ENGINE=django.db.backends.postgresql
SQL_DATABASE=postgres_db
SQL_USER=hello_django
SQL_PASSWORD=hello_django
SQL_HOST=db
SQL_PORT=5432
DATABASE=postgres
```

__Optional: Set production environment variables for testing (e.g. sendinblue email backend)__

```
EMAIL_BACKEND=anymail.backends.sendinblue.EmailBackend
DEFAULT_FROM_EMAIL=info@example.com
EMAIL_API_URL=https://api.sendinblue.com/v3/
EMAIL_API_KEY='VeRySeCrEtKeY'
```

## .env.prod


This file contains the same variables as `.env.dev`, with values tailored to the production environment.

```
DEBUG=0                                                               # Do not show debug information
SECRET_KEY=<long string of random characters>                         # Use a strong secret key, e.g., '$k=^hblb*o5^gbvu2!n58zx(&-_=1tnu3)p)#-txrf&49p3-pv'
DJANGO_ALLOWED_HOSTS=localhost 127.0.0.1 [::1] <additional hosts>     # Include the hostname of the production server (e.g., example.com)
SQL_ENGINE=django.db.backends.postgresql                              # Using PostgreSQL; see https://docs.djangoproject.com/en/3.1/ref/databases/ for more information
SQL_DATABASE=<DATABASE-NAME>                                          # Database name, should match the one in `.env.prod.db`
SQL_USER=<DATABASE-USER>                                              # Database username, should match the one in `.env.prod.db`
SQL_PASSWORD=<DATABASE-PASSWORD>                                      # Database password, should match the one in `.env.prod.db`
SQL_HOST=db                                                           # Default database host
SQL_PORT=5432                                                         # Default database port
DATABASE=postgres                                                     # Database type, should align with the SQL_ENGINE variable
EMAIL_BACKEND=<email backend>                                         # See https://pypi.org/project/django-anymail/ for supported backends
DEFAULT_FROM_EMAIL=<from address>                                     # This address will appear as the sender when users receive emails from Tweeti
EMAIL_API_URL=<API URL>                                               # API URL from your email provider
EMAIL_API_KEY=<API key>                                               # API key from your email provider
```

## .env.prod.db

```
POSTGRES_USER=<DATABASE-USER>
POSTGRES_PASSWORD=<DATABASE-PASSWORD>
POSTGRES_DB=<DATABASE-NAME>
```



\newpage{}


CHANGELOG.md
# [Unreleased]

## Added
- Initial release of Tweeti, a web-based tool for labeling tweets and short messages.
- Support for labeling on multiple devices, including laptops, phones, and tablets.
- Shortcut functionality to speed up the labeling process.
- Environment variable configurations for both development and production environments.
- Docker and Docker Compose setup for easy deployment in both development and production environments.
- Django back-end with REST API support.
- Front-end built using JavaScript and jQuery, with Material Design for easy component styling.
- Documentation for contributing, setting up environment variables, and deploying via Docker.

## Fixed
N/A (initial release)

## Changed
N/A (initial release)


tweet_label_tool/settings.py
"""
Django settings for tweet_label_tool project.

Generated by 'django-admin startproject' using Django 3.1.3.

For more information on this file, see
https://docs.djangoproject.com/en/3.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.1/ref/settings/
"""
import os
from pathlib import Path


# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.1/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
# SECRET_KEY = 'b6&j=x^t0-sx!5_phubv98^)^y5x)zt2&!i!#n$%m7_q!g^ko3'
SECRET_KEY = os.environ.get("SECRET_KEY")

# SECURITY WARNING: don't run with debug turned on in production!
# DEBUG = True
DEBUG = int(os.environ.get("DEBUG", default=0))


# 'DJANGO_ALLOWED_HOSTS' should be a single string of hosts with a space between each.
# For example: 'DJANGO_ALLOWED_HOSTS=localhost 127.0.0.1 [::1]'
ALLOWED_HOSTS = os.environ.get("DJANGO_ALLOWED_HOSTS").split(" ")


# Application definition
INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.admindocs',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'labeling', 'rest_framework',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]

ROOT_URLCONF = 'tweet_label_tool.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [os.path.join(BASE_DIR, 'templates')],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'tweet_label_tool.wsgi.application'


# Database
# https://docs.djangoproject.com/en/3.1/ref/settings/#databases
DATABASES = {
    "default": {
        "ENGINE": os.environ.get("SQL_ENGINE", "django.db.backends.sqlite3"),
        "NAME": os.environ.get("SQL_DATABASE", BASE_DIR / "db.sqlite3"),
        "USER": os.environ.get("SQL_USER", "user"),
        "PASSWORD": os.environ.get("SQL_PASSWORD", "password"),
        "HOST": os.environ.get("SQL_HOST", "localhost"),
        "PORT": os.environ.get("SQL_PORT", "5432"),
    }
}


# Password validation
# https://docs.djangoproject.com/en/3.1/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/3.1/topics/i18n/
LANGUAGE_CODE = 'en-us'
TIME_ZONE = 'UTC'
USE_I18N = True
USE_L10N = True
USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.1/howto/static-files/
STATIC_ROOT = os.path.join(BASE_DIR, "static")
STATIC_URL = '/static/'


# Redirect to home URL after login (Default redirects to /accounts/profile/)
LOGIN_REDIRECT_URL = '/'


# set email backend see: https://docs.djangoproject.com/en/3.1/topics/email/
default_email_backend = 'django.core.mail.backends.filebased.EmailBackend'
EMAIL_FILE_PATH = os.path.join(BASE_DIR, 'tmp')
EMAIL_BACKEND = os.environ.get("EMAIL_BACKEND", default_email_backend)
DEFAULT_FROM_EMAIL = os.environ.get("DEFAULT_FROM_EMAIL")


# Depending on the selected email backend, update environmental variables
# Use .env.dev and .env.prod to define the environment variables
# For more information on Django see: https://docs.djangoproject.com/en/3.1/topics/email/#topic-email-backends
# For more information on anymail see: https://pypi.org/project/django-anymail/
ANYMAIL = {
    "API_URL": os.environ.get("EMAIL_API_URL"),
    "API_KEY": os.environ.get("EMAIL_API_KEY"),
}


# upload parameters
# change this value when large file are uploaded
DATA_UPLOAD_MAX_NUMBER_FIELDS = 10240
